Dockerfile

```diff
@@ -20,18 +21,20 @@
 #    docker run -p 8888:8888 -v "$(pwd)":/tmp quay.io/fenicsproject/dolfinx:notebook
 #
 
-ARG BUILD_THREADS=1
 ARG PYBIND11_VERSION=2.2.4
 ARG PETSC_VERSION=3.10.2
 ARG SLEPC_VERSION=3.10.1
 ARG PETSC4PY_VERSION=3.10.0
 ARG SLEPC4PY_VERSION=3.10.0
 
+ARG BUILD_THREADS=1
```

Comment:
Should this be `PETSC_BUILD_THREADS` since we only use it for PETSc/SLEPc.

DOLFIN build with ninja, which has some smarts for parallel builds.

Reply:
Well, ninja allocates cpus + 2 threads. However, there is quite some criticism of that approach here: https://github.com/ninja-build/ninja/issues/1441 and https://github.com/ninja-build/ninja/pull/1399 Perhaps it is not so smart!

For optimal control it might be good to add BUILD_THREADS to ninja as well.

Reply:
I think the correct solution is to let both ninja and PETSc do their default thing (no flags) and let BUILD_THREADS override -j if set. It shouldn't be 1 by default.

Reply:
I would rather let the build systems do there own things, unless we're seeing a problem that needs fixing.